source/common/router/router.cc

```diff
@@ -191,6 +191,17 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   }
 
   timeout_ = FilterUtility::finalTimeout(*route_entry_, headers);
+
+  // If this header is set with any value, use an alternate response code on timeout
```

Comment:
I guess we can do that - any downside to allowing an arbitrary response code here?